src/main/java/com/github/packageurl/PackageURL.java

```diff
@@ -298,7 +300,7 @@ private String validateName(final String value) throws MalformedPackageURLExcept
                 temp = value.toLowerCase();
                 break;
             case StandardTypes.PYPI:
-                temp = value.replaceAll("_", "-").toLowerCase();
+                temp = value.replace('_', '-').toLowerCase();
```

Comment:
```suggestion
                temp = value.replace('_', '-').toLowerCase(Locale.ROOT);
```
Same as above.

Reply:
All these are fixed in a separate pull request #164 or #169.